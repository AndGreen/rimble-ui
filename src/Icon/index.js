import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { Icon as RmdIcon } from 'rmdi';
import theme from '../theme';
import Box from '../Box';

const StyledBox = styled('div')``;

const Icon = React.forwardRef((props, ref) => (
  <StyledBox ref={ref}>
    <RmdIcon.default {...props} />
  </StyledBox>
));

// RmdIcon.default.defaultProps = {
//   theme,
//   color: 'inherit'
// };

StyledBox.defaultProps = {
  theme,
};

Icon.defaultProps = {
  // theme,
  display: 'block',
  size: '24px',
};

Icon.propTypes = {
  /**
   * Sets which icon to display
   */
  name: PropTypes.string.isRequired,
  /**
   * Sets the element display property
   */
  display: PropTypes.string,
  /**
   * Sets the width and height in pixels
   */
  size: PropTypes.string,
  theme: PropTypes.object,
};

Icon.displayName = 'Icon';

export default Icon;
