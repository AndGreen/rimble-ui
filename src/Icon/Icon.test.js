import Icon from './'

describe('Icon component sanity', () => {
  it('has name', () => {
    expect(Icon.displayName).toBe('Icon')
  })

  it('matches default snapshot', () => {
    const component = render(<Icon />)
    expect(component).toMatchSnapshot()
  })
})
