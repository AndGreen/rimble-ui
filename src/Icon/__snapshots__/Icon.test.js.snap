// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component sanity matches default snapshot 1`] = `null`;
